src/data/listings.js

```diff
@@ -0,0 +1,41 @@
+import { ObjectId } from "mongodb";
+import { DataBaseException, NotFoundException, InvalidInputException } from "../utils/exceptions.js";
+import Validator from "../utils/validator.js";
+import { listings } from "./init.js";
+
+const getListingByUser = async (userId) => {
+const validId = Validator.validateId(userId);
+try {
+const db = await listings();
+const result = await db.find({ sellerId: new ObjectId(validId) }).toArray();
+if (!result) throw new NotFoundException(Listings not found for user ${userId});
+return result;
+} catch (e) {
+throw new DataBaseException(e.message);
+}
+};
+
+const createListing = async (sellerId, item) => {
+const validSellerId = Validator.validateId(sellerId);
+const validItem = Validator.validateItem(item);

```

Comment:
Is there a validateItem function on the Validator? I created a validateListing function, but it's empty. You will have to implement it.